test/format/compact.c

```diff
@@ -28,6 +28,68 @@
 
 #include "format.h"
 
+/*
+ * background_compact --
+ *     Periodically enable/disable the background compaction thread.
+ */
+WT_THREAD_RET
+background_compact(void *arg)
+{
+    SAP sap;
+    WT_CONNECTION *conn;
+    WT_DECL_RET;
+    WT_SESSION *session;
+    u_int period;
+    char config_buf[128];
+
+    (void)(arg);
+
+    conn = g.wts_conn;
+
+    /* Open a session. */
+    memset(&sap, 0, sizeof(sap));
+    wt_wrap_open_session(conn, &sap, NULL, &session);
+
+    /*
+     * Start the background compaction server at somewhere under 15 seconds, and then enable/disable
+     * it every 10 minutes.
```

Comment:
The only doubt I have is what is the default time the test/format runs. if it is less than 10 minutes, we need to adjust this one also. Otherwise, we never turn it off during the test/format run.

Reply:
I agree with you. We could randomise this value, we could make it happen N times during a run or simply make sure it is performed at least once during a run.

Reply:
As discussed offline, we will leave it this way for now. The code path may not be triggered with short test/format runs. However, with our longer ones that run for several hours, this will be triggered and tested.